### ISSUE TITLE
rgxp->url encodes even when decodeEntities is set to true

### ISSUE THREAD

**Issue by Toflar** [contao/contao#3401]
**Affected version(s)**

All

**Description**

If I configure a field with `eval->rgxp = 'url'` as well as `eval->decodeEntities = 'true'` I end up having encoded `&amp;`'s in the database.

This should probably be checked here https://github.com/contao/contao/blob/bdc6e30cf50421bf3ec98d67b0b21ddd5ffa950d/core-bundle/src/Resources/contao/library/Contao/Widget.php#L989

**Comment by ausi** [comment]
We could probably add something like this:

```php
if ($this->decodeEntities) {
	$varInput = StringUtil::decodeEntities($varInput);
}
```

**Comment by Toflar** [comment]
I can confirm this fixes the issue :)

**Comment by ausi** [comment]
Closed in favor of #3407
